### PR TITLE
refactor: use shiki as highlighter in demo snippets

### PR DESCRIPTION
**Pull request by cexbrayat**
Replaces prismjs that is no longer actively maintained and does not highlight Angular templates properly

Currently with prism

Capture d’écran 2024-10-02 à 21 53 01

With Shiki (this is one of the many themes possible, feel free to pick another one):

Capture d’écran 2024-10-02 à 21 53 06

### CODE REVIEW

File: projects/demo/src/app/highlighter.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import { HighlighterService } from './highlighter.service';
+
+describe('HighlighterService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+  });
+
+  it('should highlight html', async () => {
+    const prismService = TestBed.inject(HighlighterService);
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.spec.ts:11]: Rename prismService

File: projects/demo/src/app/highlighter.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    expect(prismService).toBeTruthy();
+    const result = await prismService.highlight('<html><body>Hello</body></html>', 'html');
+    expect(result).toMatch('>Hello</span>');
+  });
+
+  it('should highlight typescript', async () => {
+    const prismService = TestBed.inject(HighlighterService);
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.spec.ts:18]: Rename prismService

File: projects/demo/src/app/highlighter.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('HighlighterService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+  });
+
+  it('should highlight html', async () => {
+    const prismService = TestBed.inject(HighlighterService);
+    expect(prismService).toBeTruthy();
+    const result = await prismService.highlight('<html><body>Hello</body></html>', 'html');
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.spec.ts:13]: Maybe use the language we actually use in reality (angular-ts, angular-html)

File: projects/demo/src/app/highlighter.service.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { codeToHtml } from 'shiki';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HighlighterService {
+  async highlight(code: string, lang: string): Promise<string> {
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.ts:8]: nit: use a union type for the lang

File: projects/demo/src/app/highlighter.service.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    return codeToHtml(code.toString().trim(), {
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.ts:9]: Why the toString()?

File: projects/demo/src/app/highlighter.service.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+      lang,
+      theme: 'catppuccin-frappe'
```
Comment (jnizet) [inline projects/demo/src/app/highlighter.service.ts:11]: ultra-nit: I prefer a white or light theme because I'm daltonian, but it's not a big deal.

**Review by jnizet — CHANGES_REQUESTED** [review]

File: projects/demo/src/app/highlighter.service.ts (hunk 5 above)
Comment (cexbrayat) [reply projects/demo/src/app/highlighter.service.ts:9]: It was in the prism service, I pasted without thinking 🤷👍

**Pull request merged**